samples/matlab_experimental/ignite.m

```diff
@@ -21,21 +21,80 @@
 
     gas.TPX = {1001.0, OneAtm, 'H2:2,O2:1,N2:4'};
     gas.basis = 'mass';
-    y0 = [gas.U
-          1.0 / gas.massDensity
-          gas.Y'];
+    y0 = [gas.U, 1.0 / gas.massDensity, gas.Y];
 
-    time_interval = [0 0.001];
+    time_interval = [0, 0.001];
     options = odeset('RelTol', 1.e-5, 'AbsTol', 1.e-12, 'Stats', 'on');
 
     t0 = cputime;
     out = ode15s(@reactor_ode, time_interval, y0, options, gas, ...
-                @vdot, @area, @heatflux);
+                 @vdot, @area, @heatflux);
     disp(['CPU time = ' num2str(cputime - t0)]);
 
     plotdata = output(out, gas);
 
-    %% Time-varying boundary conditions
+    toc
+end
+
+
+function dydt = reactor_ode(t, y, gas, vdot, area, heatflux)
+    %% ODE system for a generic zero-dimensional reactor
+    %
+    % Function ``REACTOR_ODE`` evaluates the system of ordinary differential equations
+    % for a zero-dimensional reactor with arbitrary heat transfer and volume change.
+    % Used in :doc:`ignite.m <ignite>`.
+    %
+    % Solution vector components:
+    %    :y(1):     Total internal energy U
+    %    :y(2):     Volume V
+    %    :y(3):     Mass of species 1
+    %    :....:
+    %    :y(nsp+2): Mass of last species
+    %
+
+    [m, n] = size(y);
+    dydt = zeros(m, n);
+
+    for j = 1:n
+        this_y = y(:, j);
+        int_energy = this_y(1);
+        vol = this_y(2);
+        masses = this_y(3:end);
+
+        % evaluate the total mass, and the specific internal energy and volume.
+        total_mass = sum(masses);
+        u_mass = int_energy / total_mass;
+        v_mass = vol / total_mass;
+
+        % set the state of the gas by specifying (u,v,{Y_k})
+        gas.Y = masses;
```

Comment:
Does access to the C++ function `Phase::setMassFractions_NoNorm` exist in the Matlab interface? It is critical to getting reasonable performance when integrating this system of equations if the Jacobian is being evaluated using finite differences (which is the case here).

Reply:
Please note that the generated CLib doesn't include the flag `norm` that the legacy CLib had.